model/installation_states.go

```diff
@@ -121,9 +121,8 @@ func validTransitionToInstallationStateCreationRequested(currentState string) bo
 func validTransitionToInstallationStateUpgradeRequested(currentState string) bool {
 	switch currentState {
 	case InstallationStateStable,
-		InstallationStateUpgradeRequested,
-		InstallationStateUpgradeInProgress,
-		InstallationStateUpgradeFailed:
+		InstallationStateUpdateRequested,
+		InstallationStateUpdateFailed:
```

Comment:
It's invalid to go from `InstallationStateStable` to `InstallationStateUpdateInProgress` now? Maybe this was always invalid and the intent is to go through `InstallationStateUpdateRequested` first? e.g. Stable -> UpdateRequested -> UpdateInProgress?

Reply:
So this change is saying that it is no longer possible to go from `InstallationStateUpdateInProgress` to `InstallationStateUpgradeRequested`. The reason for this is I am a bit concerned about the edge cases on what could be happening when the installation is being updated and then is asked to be updated again.

Although, I don't have a conclusive thought on why this should be yet, I want to be safe until we can test it.

Reply:
OK sounds good, mostly I wanted to make sure this line didn't get removed by accident. I don't think there's a reason to keep around states/state transitions that are not known to be necessary & are not tested, so this is a good change imho, thanks for explaining.